document/src/main/java/com/yahoo/document/DocumentUpdate.java

```diff
@@ -116,6 +119,35 @@ public DocumentUpdate applyTo(Document doc) {
         return this;
     }
 
+    /**
+     * Will prune away any field update that will not modify any field in the document.
+     * @param doc
```

Comment:
Drop this line if you're not going to add doc.

Reply:
Fixed